src/connector/src/sink/clickhouse.rs

```diff
@@ -222,20 +224,17 @@ impl Sink for ClickHouseSink {
 
         // check reachability
         let client = self.config.common.build_client()?;
-        let query_column = "select distinct ?fields from system.columns where database = ? and table = ? order by ?".to_string();
-        let clickhouse_column = client
-            .query(&query_column)
-            .bind(self.config.common.database.clone())
-            .bind(self.config.common.table.clone())
-            .bind("position")
-            .fetch_all::<SystemColumn>()
-            .await?;
-        if clickhouse_column.is_empty() {
+
+        let (clickhouse_column, sign_name) =
+            query_column_engin_from_ck(client, &self.config).await?;
+
+        if !self.is_append_only && sign_name.is_none() {
```

Comment:
Instead of checking whether sign_name is none or not, we'd better check whether the engine name is within a list of engines that support upsert.